tests/components/tado/test_config_flow.py

```diff
@@ -22,7 +23,67 @@ def _get_mock_tado_api(getMe=None):
     return mock_tado
 
 
-async def test_form(hass: HomeAssistant) -> None:
+@pytest.mark.parametrize(
+    ("exception", "error"),
+    [
+        (KeyError, "invalid_auth"),
+        (RuntimeError, "cannot_connect"),
+        (ValueError, "unknown"),
+    ],
+)
+async def test_form_exceptions(
+    hass: HomeAssistant, exception: Exception, error: str
+) -> None:
+    """Test we handle Exception."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "homeassistant.components.tado.config_flow.Tado",
+        side_effect=exception,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {"username": "test-username", "password": "test-password"},
+        )
+
+    assert result2["type"] == "form"
+    assert result2["errors"] == {"base": error}
+
+
+async def test_options_flow(hass: HomeAssistant) -> None:
+    """Test config flow options."""
+    entry = MockConfigEntry(domain=DOMAIN, data={"username": "test-username"})
+    entry.add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == "form"
+    assert result["errors"] == {}
+
+    with patch(
+        "homeassistant.components.tado.async_setup_entry",
+        return_value=True,
+    ):
+        await hass.config_entries.async_setup(entry.entry_id)
+        await hass.async_block_till_done()
+
+    result = await hass.config_entries.options.async_init(
+        entry.entry_id, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == "form"
+    assert result["step_id"] == "init"
+
+    result2 = await hass.config_entries.options.async_configure(
+        result["flow_id"],
+        user_input={},
+    )
+    assert result2["type"] == "create_entry"
```

Comment:
Should we test if the data that's inputted is saved correctly?

Reply:
This test is written to get a 100% coverage and to be sure it's hitting line 132 in `config_flow.py`:

```
async def async_step_init(
        self, user_input: dict[str, Any] | None = None
    ) -> FlowResult:
        """Handle options flow."""
        if user_input is not None:
            return self.async_create_entry(title="", data=user_input)
```

If the user_input is set, it should attempt to create a new entry.

Reply:
Don't get me wrong, 100% test coverage is nice, but you also want to test something, not only touch the code.

Currently the test emulates opening the options flow and directly clicking save. Wouldn't it be a better test if we changed a setting and then checked if that setting was processed successfully?

Reply:
You're right, I've added a mainingful scenario that will happen when hitting this line.